cmd/factory.go

```diff
@@ -0,0 +1,24 @@
+package cmd
+
+import (
+	"context"
+
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/axiomhq/axiom-go/axiom"
+)
+
+// Factory provides access to the logger, Axiom client, etc.
+type Factory struct {
```

Comment:
Can we find a better name? It's not producing anything, so it's not a factory, is it?

Why is the `CancelFunc` attached here? I'd prefer having `logger` and `client` as arguments for now, we can always make it nicer if we have too many.

Reply:
Agreed.

Reply:
But let my point out: I like that the user doesn't have to write out all the dependencies and import the packages directly. But I guess that isn't to important with just two parameters instead of the `Factory` one.